server/plugin/command.go

```diff
@@ -501,6 +562,173 @@ func (p *Plugin) handleIssue(_ *plugin.Context, args *model.CommandArgs, paramet
 	}
 }
 
+func (p *Plugin) handleWebhookAdd(ctx context.Context, parameters []string, baseURL string, githubClient *github.Client) string {
+	if len(parameters) < 1 {
+		return "Invalid parameter for add command, provide repo details in `[owner/repo]` format."
+	}
+
+	owner, repo := parseOwnerAndRepo(parameters[0], p.getBaseURL())
+
+	siteURL := *p.API.GetConfig().ServiceSettings.SiteURL + "/plugins/" + Manifest.Id + "/webhook"
+	var hook github.Hook
+	var config = make(map[string]interface{})
+	config["secret"] = p.getConfiguration().WebhookSecret
+	if len(parameters) >= 1 {
+		parameters = parameters[1:]
+		for _, val := range parameters {
+			switch {
+			case strings.Contains(val, "https://"):
+				config["url"] = val
+			case strings.Contains(val, "true") || strings.Contains(val, "false"):
+				insecureVal, _ := strconv.ParseBool(val)
+				config["insecure_ssl"] = insecureVal
+			case strings.Contains(val, "application/json") || strings.Contains(val, "x-www-form-urlencoded"):
+				config["content_type"] = val
+			default:
+				if strings.Contains(val, "*") && len(val) == 1 {
+					hook.Events = []string{"*"}
+					break
+				}
+				if err := p.CheckOptionsValid(strings.Split(val, ",")); err != nil {
+					return err.Error()
+				}
+				hook.Events = strings.Split(val, ",")
+			}
+		}
+	}
+	if _, exists := config["url"]; !exists {
+		config["url"] = siteURL
+	}
+	if _, exists := config["insecure_ssl"]; !exists {
+		config["insecure_ssl"] = false
+	}
+	if _, exists := config["content_type"]; !exists {
+		config["content_type"] = "x-www-form-urlencoded"
+	}
+	if len(hook.Events) == 0 {
+		hook.Events = []string{"*"}
+	}
+	hook.Config = config
+	githubHook, _, err := p.CreateHook(ctx, githubClient, owner, repo, hook)
+	if err != nil {
+		return err.Error()
+	}
+
+	hookDetails, _, err := p.GetHook(ctx, githubClient, owner, repo, *githubHook.ID)
+	if err != nil {
+		return err.Error()
+	}
+
+	hookURL := baseURL + owner
+	if repo != "" {
+		hookURL += "/" + repo
+	}
+	hookURL += "/settings/hooks/"
+	txt := "Webhook Created Successfully \n"
+	hookID := strconv.Itoa(int(*githubHook.ID))
+	txt += fmt.Sprintf(" *  [%s](%s) :  -  %s \n", hookID, hookURL+hookID, strings.Join(hookDetails.Events, " , "))
```

Comment:
0/5 what do you think of this approach?

```suggestion
	txt := "Webhook Created Successfully \n"

	hookURL += "/settings/hooks"
	events := strings.Join(hookDetails.Events, " , ")
	txt += fmt.Sprintf(" *  [%s](%s/%v) :  -  %s \n", hookID, hookURL, *githubHook.ID, events)
```